### DISCUSSION TITLE
How to contribute

### DISCUSSION THREAD

**Discussion by kingArch** [open-meteo/open-meteo#608] [Q&A]
Hey I have been using your API for a while now and I am currently back in school doing my master's degree. I have chosen this GitHub project as a base for my work and was wondering if you can give me a clearer idea on the contribution process. (I basically need to do a BPMN representation of your contribution flow, would be more than happy to share it afterwards.)

I was wondering if you can answer some (or all) of these questions
- Is a PR required to submit features and fix bugs ?
- How many reviewers are required ?
- What happens if a reviewer disapproves of the work done ?
- Is anyone tagged as a mandatory reviewer ?
- What is the life cycle of an issue after it is reported ?
- How are deployments managed ?

Thank you so much in advance, I have spent hours looking for github projects that met the course criteria for selection, but I would really like to use this as a base, I can also submit a CONTRIBUTING.md file for review with the details given in this answer.

**Comment by patrick-zippenfenig** [comment]
Hi, thanks for asking! The Open-Meteo project is still rather small with a limited amount of contributors, so this might not be the ideal as a study subject.

> Is a PR required to submit features and fix bugs ?

Yes, external contributors must submit PRs for any changes. Myself as the only core contributor, can directly push bug fixes (not the best practice as soon as more contributors may join). For larger feature request, PRs must be used in any case.

> How many reviewers are required ?

There is no fixed review process. For external contributions I review any changes.

> What happens if a reviewer disapproves of the work done ?

For larger changes an issue ticket should be opened beforehand to describe the intented work. Feedback will be provided to find a good solution for the problem at hand. In case still does not fit criteria, the PR might be closed.

> Is anyone tagged as a mandatory reviewer ?

As I am the only core contributor, I have to review all changes.

> What is the life cycle of an issue after it is reported ?

Issues get review and commented on by me or others and ideally provide a solution. If the issue cannot be fixed right now, it stays open for later. This may take months or even years depending on the severity of the bug or complexity of the feature request.

> How are deployments managed ?

Releases are done using git tags and GitHub Actions to compile the applications. Afterwards Docker and Ubuntu packages are generated.
